sys-kernel/gentoo-kernel/gentoo-kernel-6.3.6-r1.ebuild

```diff
@@ -136,5 +136,60 @@ src_prepare() {
 		merge_configs+=( "${dist_conf_path}/big-endian.config" )
 	fi
 
+	if use modules-sign; then
+		: "${MODULES_SIGN_HASH:=sha512}"
+		cat <<-EOF > "${WORKDIR}/modules-sign.config" || die
+			## Enable module signing
+			CONFIG_MODULE_SIG=y
+			CONFIG_MODULE_SIG_ALL=y
+			CONFIG_MODULE_SIG_FORCE=y
+			CONFIG_MODULE_SIG_${MODULES_SIGN_HASH^^}=y
+		EOF
+		if [[ -n "${MODULES_SIGN_KEY}" ]]; then
+			if [[ -e "${MODULES_SIGN_KEY}" ]]; then
+				echo "CONFIG_MODULE_SIG_KEY=\"${MODULES_SIGN_KEY}\"" \
+					>> "${WORKDIR}/modules-sign.config"
+			else
+				die "MODULES_SIGN_KEY=${MODULES_SIGN_KEY} not found!"
+			fi
+		fi
+		merge_configs+=( "${WORKDIR}/modules-sign.config" )
+	fi
+
 	kernel-build_merge_configs "${merge_configs[@]}"
 }
+
+src_install() {
+	if use modules-sign; then
+		# Use the kernel build system to strip, this ensures the modules
+		# are stripped *before* they are signed, preserving the signature
+		if use strip; then
+			export INSTALL_MOD_STRIP="--strip-unneeded"
+		fi
```

Comment:
eclass is still passing STRIP=":" so I doubt(?) stripping is working (haven't tried). Unsure if changing that would cause the build system to strip other binaries though (like objtool, sign-file, and such), which would be unwanted.

Also nice catch on =strip-unneeded, hadn't noticed it used the contents if `!= 1` and thought it was just a boolean. Given portage been using unneeded for ages it should be fine over the default strip-debug anyway (used in linux-mod-r1 too).

Reply:
Yes it appears the stripping was not actually working (the log shows it tries, but evidently it silently fails). If I change `STRIP=":"` to `STRIP="$(tc-getSTRIP)"` the file size of the modules is noticeably smaller. I don't get any pre-strip QA warnings (with or without `USE="modules-sign"`) from portage so perhaps it is safe to make this change. 

@mgorny do you know what the rationale is behind `STRIP=":"` (if any)? Can we safely change it?

Reply:
>rationale

Well, it's a normal thing to do when not restricting stripping. But if variable is only used for modules and we `dostrip -x /lib/modules` then there's not really need for it.

On another note, just noticed, imo this block shouldn't be behind USE=modules-sign. It's also an issue when compression is enabled (aka you end up compressing modules still with symbols, and then portage can't strip them). linux-mod-r1 always relies on USE=strip and doesn't let portage handle it regardless, I don't think keeping a path to let portage do it is useful for kernel modules.

Reply:
> On another note, just noticed, imo this block shouldn't be behind USE=modules-sign. It's also an issue when compression is enabled (aka you end up compressing modules still with symbols, and then portage can't strip them). linux-mod-r1 always relies on USE=strip and doesn't let portage handle it regardless, I don't think keeping a path to let portage do it is useful for kernel modules.

To add to this, maybe IUSE=+strip would actually fit better in the eclass than gentoo-kernel ebuilds. Albeit the introduction of the IUSE does annoying cause rebuilds.

Reply:
> It's also an issue when compression is enabled (aka you end up compressing modules still with symbols, and then portage can't strip them).

This is a very good point, we can hit two birds with one stone here. I've moved the `use strip` logic into the eclass, and it is now also used if we don't want to sign modules.